lnwallet/channel.go

```diff
@@ -692,10 +695,21 @@ func (lc *LightningChannel) extractPayDescs(feeRate chainfee.SatPerKWeight,
 
 		htlc := htlc
 
+		auxLeaf := fn.MapOption(
+			func(l CommitAuxLeaves) input.AuxTapLeaf {
+				leaves := l.OutgoingHtlcLeaves
+				if htlc.Incoming {
+					leaves = l.IncomingHtlcLeaves
+				}
```

Comment:
This pattern gives rise to the idea that you should pack these things into [`Duals`](https://github.com/lightningnetwork/lnd/pull/8996) and then use a `ChannelParty` to index into the Dual. This will reduce these 4 lines to a one-liner that is simpler and less error prone due to single-assignment.

Reply:
I agree to use `Dual` for things like local and remote balance or local and remote keys. But incoming/outgoing IMO doesn't directly map to local/remote. AFAIK incoming/outgoing is always static as seen from the node processing the commit (but independent of **whose** commit is being processed).

Reply:
if something is incoming it means that the origin is the remote and that the destination is local. So you could frame as "htlcLeavesOrigins" or "htlcLeavesDestinations". The mapping is definitely sound, it's just a matter of whether you want to use the tools.